# try catch문
> 실행할 코드블럭을 표시하고 예외(exception)가 발생(throw)할 경우의 응답을 지정한다.
> try 선언의 구성은 하나 혹은 그 이상의 선언을 포함한 try 블록 및 catch 항목이나 finally 항목 중  
> 최소한 하나 혹은 둘 다 포함하여 이루어진다. 즉, try선언에는 세 가지 형식이 존재한다.

## 문법
```js
try {
  // 실행될 선언들
} catch (e) { // e : catch 블록과 관련된 예외 객체를 담기 위한 식별자
  // try 블록에서 예외가 발생했을 때 실행될 선언들
} finally {
  // try 선언이 완료된 후 실행될 선언들. 이 선언들은 예외 발생 여부와 상관없이 실행된다.
}
```
#### 형식
1. `try...catch`
2. `try...finally`
3. `try...catch...finally`

> catch 블록은 try 블록 안에서 얘외가 발생하는 경우에 무엇을 할 지 명시하는 코드를 포함한다.  
> try 블록 또는 try 블록 내에서 호출된 함수 내의 명령문이 예외를 throw 하면 제어가 catch블록으로 이동한다.  
> try 블록에 예외가 발생하지 않으면 catch 블록을 건너뛴다. finally 블록은 try 블록과 catch 블록이 실행을  
> 마친 후에 항상 실행된다. try에서 예외가 발생했는지에 상관없이 항상 실행된다.

